core/src/main/scala-3.0.0-RC3/cats/derived/foldable.scala

```diff
@@ -0,0 +1,59 @@
+package cats.derived
+
+import cats.{Eval, Foldable}
+import shapeless3.deriving.{K1, Continue}
+
+object foldable extends FoldableDerivation
+
+trait ProductFoldable[T[x[_]] <: Foldable[x], F[_]] extends Foldable[F]:
+
+  val inst: K1.ProductInstances[T, F]
```

Comment:
Hmm I realise now we have a choice - classes, traits, traits with parameters (I think the last one will be most flexible)